Gems/LmbrCentral/Code/Tests/AxisAlignedBoxShapeTest.cpp

```diff
@@ -194,4 +198,150 @@ namespace UnitTest
         ShapeThreadsafeTest::TestShapeGetSetCallsAreThreadsafe(entity, numIterations, setDimensionFn);
     }
 
+    TEST_F(AxisAlignedBoxShapeTest, BoxWithTranslationRotationScaleAndTranslationOffsetHasCorrectRayIntersection)
+    {
+        AZ::Entity entity;
+        CreateAxisAlignedBox(
+            AZ::Transform(AZ::Vector3(3.0f, -5.0f, 2.0f), AZ::Quaternion(0.46f, 0.22f, 0.70f, 0.50f), 2.0f),
+            AZ::Vector3(4.0f, 5.0f, 2.0f),
+            entity);
+
+        const AZ::Vector3 translationOffset(-2.0f, 3.0f, -4.0f);
+        LmbrCentral::ShapeComponentRequestsBus::Event(
+            entity.GetId(), &LmbrCentral::ShapeComponentRequestsBus::Events::SetTranslationOffset, translationOffset);
+
+        bool rayHit = false;
+        float distance;
+        LmbrCentral::ShapeComponentRequestsBus::EventResult(
+            rayHit, entity.GetId(), &LmbrCentral::ShapeComponentRequests::IntersectRay, AZ::Vector3(2.0f, -10.0f, -6.0f),
+            AZ::Vector3(0.0f, 1.0f, 0.0f), distance);
+
+        EXPECT_TRUE(rayHit);
+        EXPECT_NEAR(distance, 6.0f, 1e-2f);
+    }
+
+    TEST_F(AxisAlignedBoxShapeTest, BoxWithTranslationRotationScaleAndTranslationOffsetHasCorrectAabb)
+    {
+        AZ::Entity entity;
+        CreateAxisAlignedBox(
+            AZ::Transform(AZ::Vector3(-2.0f, 4.0f, -2.0f), AZ::Quaternion(0.38f, 0.34f, 0.70f, 0.50f), 1.5f),
+            AZ::Vector3(3.0f, 2.0f, 6.0f),
+            entity);
+
+        const AZ::Vector3 translationOffset(4.0f, 3.0f, -1.0f);
+        LmbrCentral::ShapeComponentRequestsBus::Event(
+            entity.GetId(), &LmbrCentral::ShapeComponentRequestsBus::Events::SetTranslationOffset, translationOffset);
+
+        AZ::Aabb aabb = AZ::Aabb::CreateNull();
+        LmbrCentral::ShapeComponentRequestsBus::EventResult(
+            aabb, entity.GetId(), &LmbrCentral::ShapeComponentRequests::GetEncompassingAabb);
+
+        EXPECT_THAT(aabb.GetMin(), IsClose(AZ::Vector3(1.75f, 7.0f, -8.0f)));
+        EXPECT_THAT(aabb.GetMax(), IsClose(AZ::Vector3(6.25f, 10.0f, 1.0f)));
```

Comment:
suggestion: There's actually an IsClose for Aabbs too which you could use instead/in addition if you wanted (repeated in a few places)

Reply:
Tbh I think this is easier to read.